info/container.go

```diff
@@ -405,3 +405,36 @@ func calculateCpuUsage(prev, cur uint64) uint64 {
 	}
 	return cur - prev
 }
+
+type Percentiles struct {
+	// Indicates whether the stats are present or not.
+	Present bool `json:"present"`
+	// Average over the collected sample.
+	Mean uint64 `json:"mean"`
+	// Max seen over the collected sample.
+	Max uint64 `json:"max"`
+	// 90th percentile over the collected sample.
+	Ninety uint64 `json:"ninety"`
+}
+
+type Usage struct {
+	// Indicates amount of data available [0-100].
+	// If we have data for half a day, we'll still process DayUsage,
+	// but set PercentComplete to 50.
+	PercentComplete int32 `json:"percent_complete"`
```

Comment:
Wouldn't Percent then vary for every resource? How can a user then
interpret this value?

On Tue, Feb 3, 2015 at 5:00 PM, Rohit Jnagal <notifications@github.com>
wrote:

> In info/container.go
> <https://github.com/google/cadvisor/pull/481#discussion_r24055220>:
>
> > +type Percentiles struct {
> > +	// Indicates whether the stats are present or not.
> > +	Present bool `json:"present"`
> > +	// Average over the collected sample.
> > +	Mean uint64 `json:"mean"`
> > +	// Max seen over the collected sample.
> > +	Max uint64 `json:"max"`
> > +	// 90th percentile over the collected sample.
> > +	Ninety uint64 `json:"ninety"`
> > +}
> > +
> > +type Usage struct {
> > +	// Indicates amount of data available [0-100].
> > +	// If we have data for half a day, we'll still process DayUsage,
> > +	// but set PercentComplete to 50.
> > +	PercentComplete int32 `json:"percent_complete"`
>
> Resources available or not are tracked in Percentiles through Present.
> PercentComplete only tracks how many Usage samples we gathered irrespective
> of what type is available.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/481/files#r24055220>.
>